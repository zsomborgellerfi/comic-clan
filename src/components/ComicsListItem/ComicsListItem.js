import React from 'react';
import { connect } from 'react-redux';
import './ComicsListItem.scss'
import { Link } from 'react-router-dom';

const ComicsListItem = ({ comic }) => {
    return (
        <Link to={`/comic/${comic.name}`}>
            <div className={"ComicsListItem__container"}>
                <img className={"ComicsListItem__image"} src={comic.image} alt={comic.name} />
                <div className={"ComicsListItem__name"}>{comic.name}</div>
                <div><span className={"ComicsListItem__ownedBy"}>Owned By</span><span className={"ComicsListItem__owner"}> {comic.owner}</span></div>
            </div>
        </Link>
    );
};

export default ComicsListItem;
