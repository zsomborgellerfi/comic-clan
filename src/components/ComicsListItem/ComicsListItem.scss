@import '../../styles/variables';

.ComicsListItem {
  &__container {
    color: $text-color-1;
    display: flex;
    flex-direction: column;
    font-weight: bold;
  }
  &__image {
    width: 100%;
    max-width: 200px;
  }
  &__name {
    font-size: 22px;
    margin: 16px 0 9px;
  }
  &__ownedBy {
    color: $text-color-3;
    margin-right: 3px;
    font-weight: 500;
  }
}
