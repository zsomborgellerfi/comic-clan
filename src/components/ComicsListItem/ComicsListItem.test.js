import React from "react";
import { shallow } from "enzyme";
import ComicsListItem from "./ComicsListItem";

const props = {
  comic: {
    name: "test name",
    owner: "test owner",
    image: "test image",
  },
};
describe("ComicsListItem", () => {
  const wrapper = shallow(<ComicsListItem {...props} />);

  it("should render ComicsListItem correctly", () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("Link").props().to).toEqual("/comic/test name");
    expect(wrapper.find("img.ComicsListItem__image").props().src).toEqual(
      "test image"
    );
    expect(wrapper.find("div.ComicsListItem__name").text()).toEqual(
      "test name"
    );
  });
});
