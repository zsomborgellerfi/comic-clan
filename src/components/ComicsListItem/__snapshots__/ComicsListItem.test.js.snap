// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComicsListItem should render ComicsListItem correctly 1`] = `
<Link
  to="/comic/test name"
>
  <div
    className="ComicsListItem__container"
  >
    <img
      alt="test name"
      className="ComicsListItem__image"
      src="test image"
    />
    <div
      className="ComicsListItem__name"
    >
      test name
    </div>
    <div>
      <span
        className="ComicsListItem__ownedBy"
      >
        Owned By
      </span>
      <span
        className="ComicsListItem__owner"
      >
         
        test owner
      </span>
    </div>
  </div>
</Link>
`;
